corehq/apps/reports/standard/users/reports.py

```diff
@@ -179,15 +180,39 @@ def _html_list(changes, unstyled=True):
 
     properties = UserHistoryReport.get_primary_properties(domain)
     properties.pop("user_data", None)
-    primary_changes = {
-        properties.get(key, key): value for key, value in changes.items()
-        if key in properties
-    }
-    more_count = len(changes) - len(primary_changes)
+    primary_changes = {}
+
+    for key, value in changes.items():
+        if key in properties:
+            if key == 'commcare_location_id':
+                primary_changes[_("primary location (mobile users only)")] = cached_location_id_to_name(value)
+            if key == 'location_id':
+                primary_changes["location"] = cached_location_id_to_name(value)
+            else:
+                primary_changes[key] = value
+
+    all_changes = {}
+
+    for key, value in changes.items():
```

Comment:
Did you try combining these two loops into one that populates both `primary_changes` and `all_changes`? I'd think that would reduce duplication.

Another thing to consider to DRY up this code is to pull the commcare_location_id/location_id/else logic into a helper function.

Reply:
Absolutely, both of those points make perfect sense. I'll make those changes.

Reply:
I agree with Jenny - and I think part of the way to help with that is what I described above, where the `properties` dict controls how keys are displayed, and this code here controls the values (rather than mixing the two)